WordPress/src/main/java/org/wordpress/android/push/NotificationsProcessingService.java

```diff
@@ -212,12 +258,15 @@ private void buildNoteFromJSONObject(JSONObject jsonObject) {
 
         private void performRequestedAction(){
             if (mNote != null) {
-                if (mActionType.equals(ARG_ACTION_LIKE)) {
-                    likeComment();
-                } else if (mActionType.equals(ARG_ACTION_APPROVE)) {
-                    approveComment();
-                } else if (mActionType.equals(ARG_ACTION_REPLY)) {
-                    replyToComment();
+                if (mActionType != null) {
+                    switch (mActionType) {
+                        case ARG_ACTION_LIKE:
+                            likeComment();
+                        case ARG_ACTION_APPROVE:
+                            approveComment();
+                        case ARG_ACTION_REPLY:
+                            replyToComment();
```

Comment:
Oh my! yes, and also the `break`s are missing... no excuses, wanted to fix it rapidly. Fixing and testing now.